clang/test/SemaSYCL/device_global.cpp

```diff
@@ -0,0 +1,65 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=c++17 -sycl-std=2020 -ast-dump -verify %s | FileCheck %s
+#include "Inputs/sycl.hpp"
+
+using namespace sycl::ext::oneapi;
+
+device_global<int> glob;           // OK
+static device_global<float> static_glob;    // OK
+inline device_global<double> inline_glob;    // OK
+static const device_global<int> static_const_glob;
+
+struct Foo {
+  static device_global<char> d;  // OK
+};
+device_global<char> Foo::d;
+
+struct Bar {
+  device_global<int> e;         // ILLEGAL: non-static member variable not
+};                              // allowed
+
+struct Baz {
+ private:
+  // expected-error@+1{{'device_global' member variable 'f' is not publicly accessible from namespace scope}}
+  static device_global<int> f;  // ILLEGAL: not publicly accessible from
+};                              // namespace scope
+device_global<int> Baz::f;
+
+device_global<int[4]> not_array;        // OK
+device_global<int> h[4];        // ILLEGAL: array of "device_global" not
```

Comment:
Same question here?

Reply:
Diagnostic is not being generated for this case at the moment. Will do it in a different PR.

Reply:
Please add FIXME

Reply:
Is it ok to simply delete cases that are not currently supported?

Reply:
> Is it ok to simply delete cases that are not currently supported?

I think it makes sense.